src/Response.php

```diff
@@ -204,6 +204,15 @@ public function writeHead($status = 200, array $headers = array())
             );
         }
 
+        // assign date header if no 'date' is given, use the current time where this code is running
+        if (!isset($lower['Date'])) {
```

Comment:
Must test against `$lower['date']` (lowercase).  Also, looks like this case isn't properly tested :)